### PR TITLE
Adds support for raw latitude and longitude address inputs

### PR DESCRIPTION
**Pull request by vkkostin**
## Problem
The map component accepts street addresses and sends them to our [geocoder](https://github.com/AdaloHQ/geocoder) service, which makes a request to Google's geocode service through a [node client](https://github.com/googlemaps/google-maps-services-js), which comes back with latitude and longitude coordinates for each address. The geocode service does not accept coordinates as an input, which means the map component would not display any markers if makers were using latitude and longitude instead of street addresses for their location data.

## Solution
The map component will now differentiate between street addresses and coordinates; it will send the former to the geocoder  while the latter can bypass the geocoder entirely and be sent straight to the map. Typically a maker will only use one or the other format, but this also takes into account a mix of formats, recombining the two types of addresses before sending them to the map.

## Developer Notes
Since working with coordinates is a synchronous operation (because it bypasses the geocoder) and we set state inside the `render` hook, it resulted in an infinite render loop and crashed the component. To implement support for coordinates I extracted that functionality into the `componentDidUpdate` hook.

### CODE REVIEW

**Comment by vkkostin** [comment]
[When inputting Lat, Long into Google Maps Component, markers are not displaying on the map](https://trello.com/c/Y9ejRHiE/67-when-inputting-lat-long-into-google-maps-component-markers-are-not-displaying-on-the-map)

File: src/Map/index.js
Comment (tobyjsullivan) [inline src/Map/index.js:78]: Why is `prevProps` used in this method instead of the current props in `this.props`?

File: src/Map/index.js
Comment (tobyjsullivan) [inline src/Map/index.js:82]: Should this get called on first render as well? I.e., in `componentDidMount()`

File: src/Map/index.js
Comment (tobyjsullivan) [inline src/Map/index.js:164]: I have no reason to doubt the correctness of this logic for detecting and parsing coordinates. That said, as a reader, it's also pretty hard for me to be confident that it is correct.

What do you think about pulling this out to a small utility module, maybe `isCoordinates(string)` and `parseCoordinates(string)`, and then hitting it with some unit tests to demonstrate correctness? This might also help when it comes time to extend support to different coordinate formats in future.

File: src/Map/map.web.js
Comment (tobyjsullivan) [inline src/Map/map.web.js:56]: Using a list index as the `key` is [generally discouraged](https://robinpokorny.medium.com/index-as-a-key-is-an-anti-pattern-e0349aece318) when there's an alternative available. Would it make sense to use something like the lat/long values here instead?

**Review by tobyjsullivan — CHANGES_REQUESTED** [review]
This is a solid improvement. I've left a few questions/suggestions to consider.

File: src/Map/index.js
Comment (vkkostin) [reply src/Map/index.js:164]: We do not have a testing framework set up for components. I believe @scott-massey was working on that at some point.

File: src/Map/index.js
Comment (vkkostin) [reply src/Map/index.js:82]: No, because `markerCollection` is always falsy on-mount. I can leave a comment to that effect.

File: src/Map/index.js
Comment (vkkostin) [reply src/Map/index.js:78]: Is that not what I'm supposed to do? The documentation makes it seem like I'm supposed to the `prevProps` that this hook receives.

File: src/Map/map.web.js
Comment (vkkostin) [reply src/Map/map.web.js:56]: I can make this change. I was under the impression that using the index on it's own is discouraged.

File: src/Map/index.js
Comment (tobyjsullivan) [reply src/Map/index.js:166]: What if this logic was pulled out of the component?

Comment (tobyjsullivan) [reply src/Map/index.js:166]: I did not realize what repo this was. 👍

File: src/Map/map.web.js
Comment (tobyjsullivan) [reply src/Map/map.web.js]: That's usually the case that's discussed but my understanding is it's technically any key that is strictly a derivative of the index as all the same performance issues and potential bugs apply.

File: src/Map/index.js
Comment (tobyjsullivan) [reply src/Map/index.js:166]: (not blocking) Maybe it's still worth asking: how difficult would it be to add `jest` and write some tests for this, even if it wasn't setup on CI just yet? Would now be as good a time as any to add unit tests to this repo?

File: src/Map/index.js
Comment (tobyjsullivan) [reply src/Map/index.js:78]: Do you mean [react's documentation](https://reactjs.org/docs/react-component.html#componentdidupdate) or something else? If react, those docs show the API. Since `componentDidUpdate` get's invoked whenever props or state change, both the previous props/state and the current props/state are made available. The current props in `this.props` and the current state in `this.state`. My question is just confirming that we intend to pass the stale props to `this.loadAddresses` instead of the values from after the update. Is that correct here?

**Review by tobyjsullivan — APPROVED** [review]
My only outstanding concern is the `prevProps` thing. LGTM if that's correct 👍  (Approving now so the PR is not blocked on my required changes while I'm OOO)

File: src/Map/index.js
Comment (vkkostin) [reply src/Map/index.js:78]: Current props should be ok.

File: src/Map/index.js
Comment (vkkostin) [reply src/Map/index.js:166]: Seems like a good cool-down project. If I get it done before this is merged I can incorporate it into this card.

**Pull request merged**
